tbmalt/structures/geometry.py

```diff
@@ -175,6 +175,12 @@ def distances(self) -> Tensor:
         dist = torch.cdist(self.positions, self.positions, p=2)
         # Ensure padding area is zeroed out
         dist[self._mask_dist] = 0
+
+        # cdist bug, sometimes distances diagonal is not zero
+        idx = torch.arange(dist.shape[-1])
+        if not (dist[..., idx, idx].eq(0)).all():
+            dist[..., idx, idx] = 0.0
+
```

Comment:
This is a good catch! Please, could you send me an example where such and error occurs.